src/libs/SearchUIUtils.ts

```diff
@@ -332,10 +341,11 @@ function getAction(data: OnyxTypes.SearchResults['data'], key: string): SearchTr
     if (canIOUBePaid(report, chatReport, policy, allReportTransactions, false, chatReportRNVP, invoiceReceiverPolicy) && !hasOnlyHeldExpenses(report.reportID, allReportTransactions)) {
         return CONST.SEARCH.ACTION_TYPES.PAY;
     }
-    const hasOnlyPendingTransactions = allReportTransactions.length > 0 && allReportTransactions.every((t) => isExpensifyCardTransaction(t) && isPending(t));
+    const hasOnlyPendingCardOrScanFailTransactions =
+        allReportTransactions.length > 0 && allReportTransactions.every((t) => (isExpensifyCardTransaction(t) && isPending(t)) || (isPartialMerchant(getMerchant(t)) && isAmountMissing(t)));
 
     const isAllowedToApproveExpenseReport = isAllowedToApproveExpenseReportUtils(report, undefined, policy);
-    if (canApproveIOU(report, policy) && isAllowedToApproveExpenseReport && !hasOnlyPendingTransactions) {
+    if (canApproveIOU(report, policy) && isAllowedToApproveExpenseReport && !hasOnlyPendingCardOrScanFailTransactions) {
```

Comment:
Should we move the `isAllowedToApproveExpenseReport` and `hasOnlyPendingCardOrScanFailTransactions` logic inside `canApproveIOU`? It seems like that logic should be applied across the App and there's no need to duplicate it in several places

Reply:
I'm not sure that I follow how that's related. We should have the same `canApproveIOU` check throughout the App and just feed it different data if we're on the Inbox vs Search pages